arch/posix/CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
 # SPDX-License-Identifier: Apache-2.0
 
+execute_process(COMMAND arch OUTPUT_VARIABLE BUILD_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+if (${BUILD_ARCH} STREQUAL "aarch64")
+	zephyr_compile_options(-fPIC)
+endif ()
+
+if (${BUILD_ARCH} STREQUAL "x86_64")
```

Comment:
CMake supports `elseif(...)`

Reply:
& ending it with an else() with a warning "unknown architecture, couldn't select word-size specific compile options, expect problems" would be nice. (But I'd expect native_posix (not 64) to compile without problems in an x86 linux with the -m32 as is here)